### DISCUSSION TITLE
Announcing Temml

### DISCUSSION THREAD

**Discussion by ronkok** [KaTeX/KaTeX#2800] [Show and tell]
I've just published a new library which I call [**Temml**](https://temml.org/). It converts TeX to MathML.

It is of course a fork from KaTeX, but it also contains pretty substantial refactoring and a number of MathML bug fixes. It's coverage of LaTeX is better than KaTeX, since Temml is not tightly bound to the KaTeX fonts.

Temml is not of much use at this moment, but if the [Chromium MathML project](https://mathml.igalia.com/project/) comes through and Safari improves a bit, then Temml will be the most lightweight math rendering option available. 

@kevinbarabash The Temml [README](https://github.com/ronkok/Temml#readme) acknowledges its KaTeX roots. If KA would like further acknowledgment, let me know. And to other KaTeX contributors here, let me express my thanks. If you would like to be acknowledged by name in the Temml README, I would be willing to do so.

**Comment by edemaine** [comment]
Wow, quite a project!  As a Chrome-primary user, I'm still not on the MathML bandwagon, but maybe I'll catch up once it's finally supported.

Some meta questions:
1. Are you planning to keep Temml synchronized with KaTeX when that makes sense? Or are you imagining them diverging further over time?
2. Are there MathML bug fixes, or other features, that could be ported back to KaTeX? (Just saw this idea already floated in #2798.)

Hopefully you're not leaving the KaTeX team! Your contributions have been extremely valuable.

**Reply by ronkok** [reply]
Yes, it's still uncertain how useful MathML will turn out to be. It all depends on the quality of the browser implementations.

As to synchronization, I believe in [Postel robustness](https://en.wikipedia.org/wiki/Robustness_principle),  and I hope that any migration to Temml from KaTeX or MathJax will be fairly painless. But I don't intend to be a fanatic about it.

I did indeed find more MathML bugs, several of them. But most of them affect the visual rendering, not the screenreader quality. And if anyone is going to visually render MathML, they can get it in a more lightweight form by using Temml directly. I will probably back-port some MathML bug fixes to KaTeX, but not all of them.

My contributions to KaTeX were going to diminish whether I wrote Temml or not. I've been working for years toward feature parity with MathJax, and KaTeX is about as close to that as it is going to get.  My personal goals here are close to complete. I will naturally still be available to help if any bugs are found in my work.

**Comment by k4b7** [comment]
Cool Project!  Thanks for sharing and thanks for all of your contributions to KaTeX. ❤️
